Scripts/Runtime/Collections/UnsafeBuffer.cs

```diff
@@ -0,0 +1,550 @@
+    {
+        public static readonly int SIZE = UnsafeUtility.SizeOf<BlockInfo>();
+        public static readonly int ALIGNMENT = UnsafeUtility.AlignOf<BlockInfo>();
+
+        public byte* Data;
+        public BlockInfo* Next;
+    }
+    
+    /// <summary>
+    /// Information about a "lane" which is an indexed sub-section of the buffer that can be read from or written to.
+    /// A lane is commonly used in parallel situations so that each thread stays in its own lane.
+    /// Lane Info preserves the state of writing so that multiple jobs can write to a buffer over time.
+    /// </summary>
+    /// <remarks>
+    /// Unity will refer to this as a "foreachindex" which is a little bit confusing.
+    /// </remarks>
+    [BurstCompatible]
+    [StructLayout(LayoutKind.Sequential, Size = 64)]
+    internal unsafe struct Lane
+    {
+        public BlockInfo* FirstBlock;
+        public BlockInfo* CurrentWriterBlock;
+        public byte* WriterHead;
+        public byte* WriterEndOfBlock;
+        public int Count;
+    }
+    
+    /// <summary>
+    /// Information about the <see cref="UnsafeBuffer{T}"/> itself.
+    /// </summary>
+    [BurstCompatible]
+    internal struct BufferInfo
+    {
+        public int ElementSize;
+        public int ElementAlignment;
+        public int BlockSize;
+        public int LaneCount;
+        public Allocator Allocator;
+    }
+    
+    /// <summary>
+    /// A collection that allows for parallel reading and writing.
+    /// It looks and somewhat behaves similar to a <see cref="UnsafeStream"/> but has some key differences that make it
+    /// more advantageous.
+    ///
+    /// 1. Typed. The <see cref="UnsafeStream"/> can have anything written to it which results in arbitrarily sized and
+    /// filled "blocks". This <see cref="UnsafeBuffer{T}"/> is typed and holds an exact amount of the elements in its
+    /// "blocks".
+    /// 2. Multiple writing. The <see cref="UnsafeStream"/> will only allow for writing on a given index once. You
+    /// cannot have two jobs where one writes to index 1 and then later on in the frame a different job continues
+    /// writing on index 1. This <see cref="UnsafeBuffer{T}"/> allows for that and is geared for different jobs to fill
+    /// up the buffer.
+    /// </summary>
+    /// <typeparam name="T">The type of elements to store in the collection.</typeparam>
+    [BurstCompatible]
+    public unsafe struct UnsafeBuffer<T> : INativeDisposable
+        where T : struct
+    {
+        //See Chunk.kChunkSize (Can't access here so we redefine)
+        private const int CHUNK_SIZE = 16 * 1024;
+
+        [NativeDisableUnsafePtrRestriction] private BufferInfo* m_BufferInfo;
+        [NativeDisableUnsafePtrRestriction] private Lane* m_Lanes;
+        
+        /// <summary>
+        /// Is the collection created or not?
+        /// </summary>
+        public bool IsCreated
+        {
+            get => m_Lanes != null;
+        }
```

Comment:
Sort of but not really. 

The issue is that it's a struct so adding a private flag to check when we are disposed doesn't work. 

```
            UnsafeTypedStream<Timer> testA = new UnsafeTypedStream<Timer>(Allocator.Temp);
            //TestB is a copy of the struct of TestA however they both point to the same memory so they are the same.
            UnsafeTypedStream<Timer> testB = testA;
            //TestA's dispose is called and the memory is cleared. TestA's IsDispose flag would be true now.
            testA.Dispose();
            //This will return false. TestB's flag has not be flipped even though it is actually disposed.
            Debug.Log(testB.IsDisposed);
```

I've updated the comment to signify that `IsCreated` is true only when the memory backing the collection is valid and we have not been disposed.

Reply:
yea sorry I meant make a getter that flipped the `IsCreated` return as a convenience for developers.
Just having a comment is totally fine. Having to hunt down and infer the behaviour of IsCreated is the part I wanted to avoid.